sunpy/visualization/imageanimator.py

```diff
@@ -608,32 +585,257 @@ def _sanitize_axis_range(self, axis_range, data):
                 # If min==max or None
                 if axis_range[i] is None or axis_range[i][0] == axis_range[i][1]:
                     if i in self.slider_axes:
-                        axis_range[i] = np.linspace(0,d,d)
+                        axis_range[i] = np.linspace(0, d, d)
                     else:
                         axis_range[i] = [0, d]
                         # min max pair for slider axes should be converted
                         # to an array
                 elif i in self.slider_axes:
                     axis_range[i] = np.linspace(axis_range[i][0], axis_range[i][1], d)
 
-            #If we have a whole list of values for the axis, make sure we are a slider axis.
+            # If we have a whole list of values for the axis, make sure we are a slider axis.
             elif len(axis_range[i]) == d:
-                if i not in self.slider_axes:
-                    raise ValueError("Slider axes mis-match, non-slider axes need [min,max] pairs")
-                else:
-                    #Make sure the resulting element is a ndarray
-                    axis_range[i] = np.array(axis_range[i])
+                axis_range[i] = np.array(axis_range[i])
 
-            #panic
+            # panic
```

Comment:
It's a bit dramatic but just means if the above criteria are not satisfied raise an error which is done right after that comment.